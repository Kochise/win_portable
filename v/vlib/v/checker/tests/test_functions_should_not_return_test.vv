// ordinary functions can return whatever they like,
// since they are not called by V's testing system
// in the generated main():
fn abc() int {
	return 1
}

// should be disallowed:
fn test_returning_int() int {

}

// should be disallowed:
fn test_returning_opt() ? {
}
