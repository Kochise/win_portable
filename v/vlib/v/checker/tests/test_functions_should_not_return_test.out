vlib/v/checker/tests/test_functions_should_not_return_test.vv:9:1: error: test functions should not return anything
    7 | 
    8 | // should be disallowed:
    9 | fn test_returning_int() int {
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~
   10 | 
   11 | }
vlib/v/checker/tests/test_functions_should_not_return_test.vv:14:1: error: test functions should not return anything
   12 | 
   13 | // should be disallowed:
   14 | fn test_returning_opt() ? {
      | ~~~~~~~~~~~~~~~~~~~~~~~~~
   15 | }
